packages/flutter_tools/.analysis_options

```diff
@@ -0,0 +1,18 @@
+# A temporary file to do some validation in the flutter_tools project. This will
+# go away once we unify `flutter analyze` and analysis server based products.
+
+analyzer:
+  language:
+    enableStrictCallChecks: true
+    enableSuperMixins: true
+  errors:
+    todo: ignore
+linter:
+  rules:
+    - avoid_empty_else
+    - always_declare_return_types
+    - always_specify_types
+    - annotate_overrides
+    - sort_unnamed_constructors_first
+    - unnecessary_brace_in_string_interp
+    - unnecessary_getters_setters
```

Comment:
I notice this is a small subset of the lint rules enumerated in `flutter_analysis_options`.  Curious what the rationale is?  I would have expected them all % `public_member_api_docs`?